crates/core/src/operations/write/async_utils.rs

```diff
@@ -28,6 +28,7 @@ impl AsyncShareableBuffer {
     }
 
     /// Returns a clone of the underlying buffer as a `Vec`.
+    #[allow(dead_code)]
```

Comment:
some of these things that clippy claims were dead seemed important so I didn't actually removed them, I just added an allow. 

I am happy to remove them too if you prefer